### PR TITLE
move enum.getMaxMinValue to typesemantic, its only callsite

### PR DESCRIPTION
**Pull request by benjones**
Continue moving semantic stuff to the semantic modules

### CODE REVIEW

File: src/dmd/typesem.d [hunk 1]
```diff
@@ -4743,3 +4743,122 @@ Expression defaultInit(Type mt, const ref Loc loc)
...
+ * Lazily computes the value and caches it in maxval/minval.
+ * Reports any errors.
+ * Params:
+ *      ed = the EnumDeclaration being examined
+ *      loc = location to use for error messages
+ *      id = Id::max or Id::min
+ * Returns:
+ *      corresponding value of .max/.min
+ */
+Expression getMaxMinValue(EnumDeclaration ed, const ref Loc loc, Identifier id)
```
Comment (thewilsonator) [inline src/dmd/typesem.d:4759]: `private` if typeset is the only user

**Review by thewilsonator — APPROVED** [review]
otherwise looks good.

**Comment by thewilsonator** [comment]
Thanks!

**Comment by thewilsonator** [comment]
Hmm this fails.

**Comment by benjones** [comment]
> Hmm this fails.

Looks like I needed to use the error method, not the global error function.  I think I passed the failing fail_compile test locally, so hopefully it's all good now.

**Comment by dkorpel** [comment]
I'm a bit late, but I feel this should actually go to denum.d, because the function is touching innards of `EnumMember` (`&ed.maxval` and `ed.inuse++`) that could otherwise be made `private`. Also typesem.d is already quite large (4800 lines) and denum.d fairly small (350 lines).

**Comment by benjones** [comment]
> I'm a bit late, but I feel this should actually go to denum.d, because the function is touching innards of `EnumMember` (`&ed.maxval` and `ed.inuse++`) that could otherwise be made `private`. Also typesem.d is already quite large (4800 lines) and denum.d fairly small (350 lines).
> Edit: derp, it _was_ in denum. Note sure if this is an improvement, but eh.

The goal is to get rid of imports to semantic modules in the AST modules so they can be used by a library without dragging in semantic.

**Pull request merged**